src/async/python/async.py

```diff
@@ -0,0 +1,126 @@
+        # Could do something else here.
+        await asyncio.sleep(0)
+        await identify_task  # Block until the identify_async() call completes.
+        x_axis = x_device.get_axis(X_AXIS_NUMBER)
+
+        if Y_DEVICE_ADDRESS == X_DEVICE_ADDRESS:
+            y_axis = x_device.get_axis(Y_AXIS_NUMBER)
+        else:
+            y_device = connection.get_device(Y_DEVICE_ADDRESS)
+            await y_device.identify_async()
+            y_axis = y_device.get_axis(Y_AXIS_NUMBER)
+
+        # Home the devices and wait until done.
+        # This is an example of overlapping async commands. Order of execution is not
+        # guaranteed and with more than two or three there is a risk of a device error.
+        await asyncio.gather(x_axis.home_async(), y_axis.home_async())
+
+        # Grid scan loop
+        axes = [x_axis, y_axis]
+        async for coords in grid(X_GRID_POINTS, Y_GRID_POINTS):
+            # Here's one way of controlling the devices in parallel. This loop potentially sends the
+            # move commands and waits for the acknowledgements on different threads.
+            for index, axis in enumerate(axes):
+                position = coords[index] * GRID_SPACING
+                # Avoid the tempation to save the tasks in an array and await them as a group in the
+                # case of movement commands, as the device may not be able to consume the commands
+                # as fast as you can send them, and a system error may occur. Two or three should
+                # be safe, as in the home command above.
+                # Again, you can move the await to a later point in the loop if you want to.
+                await axis.move_absolute_async(position, Units(GRID_SPACING_UNITS), False)
+
+            # At this point the axes are moving and we may have many milliseconds or seconds to
+            # do other work (not shown).
+
+            # Now when we need to be sure the devices have stopped moving, we can wait until they
+            # are idle.
+            for axis in axes:
+                await axis.wait_until_idle_async()
+
+            # At this time the devices have all reached one of the target points and have stopped
+            # moving. This is where you could insert some other code to do something like take a
+            # picture or sample a well plate cell.
+            print(f"At point { ', '.join([str(n) for n in coords]) }")
+
+    # The connection will automatically be closed at the end of the "with" block.
+
+
+# This function could be synchronous and return just an array of coordinates, or
+# (more commonly) you could generate the coordinates in the device control loop
+# above. This is example is contrived to simulate reading the coordinates from
+# an external source.
+async def grid(x_points: int, y_points: int) -> AsyncIterator[List[int]]:
+    """Generate points in boustrophedon (serpentine, minimum-movement) order."""
```

Comment:
I am not sure if `boustrophedon` is a well known term.  If you just google it, it refers to writing and is not clear that it refers to a particular path.  May be preferable not to use the term as it may confuse more than aid.